Add Residue Charge color scheme

luk27official opened the pull request:
# Description

This PR adds a new "Residue Charge" color palette for residues. I have added it as a new palette for the "Residue Name" scheme. Only certain residues are colored as most of the residues are not charged. Histidine is colored in light blue as it might not always be charged.

References:
- https://www.rbvi.ucsf.edu/chimerax/data/electrostatics-apr2021/crm1.html
- https://pubmed.ncbi.nlm.nih.gov/26528483/

Feel free to change the codes for the colors if needed.

Including a screenshot for [6tvo](https://www.rcsb.org/structure/6tvo)

<img width="808" height="678" alt="6TVO_A_CENTERED PDB" src="https://github.com/user-attachments/assets/a4a70134-413b-47b2-a0bc-3f5e910669ed" />

## Actions

- [x] Added description of changes to the `[Unreleased]` section of `CHANGELOG.md`
- [x] Updated headers of modified files
- [x] Added my name to `package.json`'s `contributors`
- [ ] (Optional but encouraged) Improved documentation in `docs`

dsehnal commented on src/mol-theme/color/residue-name.ts:
```diff
@@ -71,7 +122,8 @@ export const ResidueNameColorThemeParams = {
         'default': PD.EmptyGroup(),
-        'custom': PD.Group(getColorMapParams(ResidueNameColors))
+        'custom': PD.Group(getColorMapParams(ResidueNameColors)),
+        'residue-charge': PD.Group(getColorMapParams(ChargedResidueColors))
```
This feels quite misplaced to me -- I would prefer this was a new color theme altogether. Basically a copy-paste of this theme, but rename it to `residue-charge` and perhaps allow alternative color schemes.

What do you think?

prihoda replied:
I agree, ideally putting it directly below/above Hydrophobicity

luk27official replied:
Yes, I didn't want to duplicate the code too much but it makes sense

> I agree, ideally putting it directly below/above Hydrophobicity

Now the options are listed alphabetically so I will keep it that way

luk27official commented:
I have addressed the comments :)

- changed colors
- moved to a separate file
- added references

<img width="1532" height="1283" alt="1TQN" src="https://github.com/user-attachments/assets/4121faf8-d02d-4852-84ce-e9d5eca22309" />
